packages/hardhat-vyper/src/compilation.ts

```diff
@@ -100,17 +100,21 @@ async function isAlreadyCompiled(
   }
 
   const contractName = pathToContractName(sourceFile);
-  const artifactPath = path.join(paths.artifacts, `${contractName}.json`);
+  const contractPath = `/contracts${sourceFile.split("/contracts")[1]}`;
+  const artifactPath = path.join(paths.artifacts, `${contractPath}/${contractName}.json`);
+
   if (!(await fsExtra.pathExists(artifactPath))) {
     return false;
   }
+  
+  const sourceFilePath = sources.find(s => s.includes(contractPath));
+  if (!sourceFilePath) {
+    return false;
+  }
```

Comment:
I don't understand why this is necessary, and why not just do

```
const lastSourcesCtime = (await fsExtra.stat(sourceFile)).ctimeMs;
```

below.

Reply:
Ok yes this is better. I was just modified the old implementation which mapped over the sources array.